adapters/adman/adman.go

```diff
@@ -0,0 +1,146 @@
+package adman
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// AdmanAdapter struct
+type AdmanAdapter struct {
+	URI string
+}
+
+// NewAdmanBidder Initializes the Bidder
+func NewAdmanBidder(endpoint string) *AdmanAdapter {
+	return &AdmanAdapter{
+		URI: endpoint,
+	}
+}
+
+type admanParams struct {
+	TagID string `json:"TagID"`
+}
+
+// MakeRequests create bid request for adman demand
+func (a *AdmanAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var admanExt openrtb_ext.ExtImpAdman
+	var err error
+
+	var adapterRequests []*adapters.RequestData
+
+	for _, imp := range request.Imp {
+		reqCopy := *request
```

Comment:
Given that `reqCopy.Imp` is the only `reqCopy` field that gets modified at every loop, I believe it is safe and a bit more optimal to make this copy outside the `for` loop.
```
 31   func (a *AdmanAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
 32       var errs []error
 33       var admanExt openrtb_ext.ExtImpAdman
 34       var err error
 35  
 36       var adapterRequests []*adapters.RequestData
 37  
    +     reqCopy := *request
 38       for _, imp := range request.Imp {
 39 -         reqCopy := *request
 40           reqCopy.Imp = []openrtb.Imp{imp}
 41  
 42   *-- 17 lines: var bidderExt adapters.ExtImpBidder--------------------------------------------------------------------------
 59           errs = append(errs, errors...)
 60       }
 61       return adapterRequests, errs
 62   }
adapters/adman/adman.go
```